pyppeteer/emulation_manager.py

```diff
@@ -17,33 +17,35 @@ def __init__(self, client: CDPSession) -> None:
         self._hasTouch = False
 
     async def emulateViewport(self, viewport: dict) -> bool:
-        """Evaluate viewport."""
-        options = dict()
+        """
+        Evaluate viewport.
+        :param viewport: dictionary which supports keys: isMobile, width, height,
+            deviceScaleFactor, isLandscape, hasTouch
+        """
         mobile = viewport.get('isMobile', False)
-        options['mobile'] = mobile
-        if 'width' in viewport:
-            options['width'] = helper.get_positive_int(viewport, 'width')
-        if 'height' in viewport:
-            options['height'] = helper.get_positive_int(viewport, 'height')
-
-        options['deviceScaleFactor'] = viewport.get('deviceScaleFactor', 1)
+        width = viewport.get('width')
+        height = viewport.get('height')
+        deviceScaleFactor = viewport.get('deviceScaleFactor', 1)
         if viewport.get('isLandscape'):
-            options['screenOrientation'] = {'angle': 90,
-                                            'type': 'landscapePrimary'}
+            screenOrientation = {'angle': 90, 'type': 'landscapePrimary'}
         else:
-            options['screenOrientation'] = {'angle': 0,
-                                            'type': 'portraitPrimary'}
+            screenOrientation = {'angle': 0, 'type': 'portraitPrimary'}
         hasTouch = viewport.get('hasTouch', False)
 
-        await self._client.send('Emulation.setDeviceMetricsOverride', options)
-        await self._client.send('Emulation.setTouchEmulationEnabled', {
-            'enabled': hasTouch,
-            'configuration': 'mobile' if mobile else 'desktop'
-        })
-
-        reloadNeeded = (self._emulatingMobile != mobile or
-                        self._hasTouch != hasTouch)
-
+        await asyncio.gather(
+            self._client.send(
+                'Emulation.setDeviceMetricsOverride', {
```

Comment:
Out of the scope of this PR, but I wonder if it'd be useful to write a script to generate enums based on [this](https://github.com/ChromeDevTools/devtools-protocol/blob/master/json/browser_protocol.json), and modify CDPSession to send the str representation of those enums. The benefit would be autocompletion from your IDE when developing. OTOH, it would (overall) add a lot of imports, and most commands are copy/pasted from the puppeteer, so it's unlikely errors would enter from *not* have devtools protocal commands as enums.

Reply:
That's a pretty cool idea that would allow easier hacking on chrome as well. In general I'm in favor of enabling enums and constants where there are strings.

Reply:
I've created issue for this idea #45